test/ModuleLoader.test.cpp

```diff
@@ -55,11 +55,19 @@ TEST(ModuleLoader, InternalModulePassed)
 
 	EXPECT_EQ("IPDropDown", moduleManager.GetModuleName(0));
 
-	// Ensure ModuleManager does not allow multiple modules with same name but different casing
-	EXPECT_NO_THROW(moduleManager.RegisterModule(static_cast<std::unique_ptr<GameModule>>(std::make_unique<DifferentCasedNameModule>())));
-	EXPECT_EQ(1u, moduleManager.Count());
-
 	EXPECT_NO_THROW(moduleManager.LoadModules());
 	EXPECT_NO_THROW(moduleManager.RunModules());
 	EXPECT_NO_THROW(moduleManager.UnloadModules());
 }
+
+TEST(ModuleLoader, RejectCaseSensitiveNames)
```

Comment:
The test name is not entirely clear and maybe a little misleading. A string by itself is not really case sensitive nor case insensitive. It's a comparison of two or more strings that can be done with regards to case sensitivity. Related, though not relevant here, is case folder versus case preserving. That would apply to how a single string is handled, and may affect how a comparison might be done. Here, registering a module name is case preserving, while comparisons are case insensitive. The phrasing for what we want to express feels a bit awkward. Maybe "RejectCaseInsensitiveDuplicateNames"?

The test itself is wonderfully clear. I like how you separated the checks into their own test case. The code is very targeted and easy to read.

Reply:
Thank you for reviewing.